wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java

```diff
@@ -482,6 +490,28 @@ private RecipeParser getRecipeParser(StageContext context)
     return new GrammarBasedParser(context.getNamespace(), new MigrateToV2(directives).migrate(), registry);
   }
 
+  private void emitDirectiveMetrics(List<Directive> directives, Metrics metrics) {
+    for (Directive directive : directives) {
+      if (directive instanceof DirectiveWithMetrics) {
+        List<DirectiveMetric> directiveMetrics = ((DirectiveWithMetrics) directive).getMetrics();
+        for (DirectiveMetric directiveMetric : directiveMetrics) {
+          Metrics child = metrics.child(directiveMetric.getTags());
+          // make this common to both count and gauge
+          child.count(directiveMetric.getMetricName(), 1);
+        }
+      }
+      emitDirectiveUsedMetric(directive.define().getDirectiveName(), metrics);
+    }
+  }
+
+  private void emitDirectiveUsedMetric(String directiveName, Metrics metrics) {
```

Comment:
Lets add some javadoc on how we are mapping the wrangler metrics into common entities of CDAP.